src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -1098,7 +1098,21 @@ address TemplateInterpreterGenerator::generate_native_entry(bool synchronized) {
     __ ld(x28, Address(xmethod, Method::native_function_offset()));
     address unsatisfied = (SharedRuntime::native_method_throw_unsatisfied_link_error_entry());
     __ mv(t1, unsatisfied);
-    __ ld(t1, Address(t1, 0));
+    if (AvoidUnalignedAccesses) {
+      __ mv(t, t1);
+      __ lhu(t1, Address(t1, 0));
+      __ lhu(t0, Address(t, 2));
+      __ slli(t0, t0, 16);
+      __ add(t1, t1, t0);
+      __ lhu(t0, Address(t, 4));
+      __ slli(t0, t0, 32);
+      __ add(t1, t1, t0);
+      __ lhu(t0, Address(t, 6));
+      __ slli(t0, t0, 48);
+      __ add(t1, t1, t0);
```

Comment:
Maybe another `load_long_misaligned` assembler function for this long sequence?

Reply:
yeah, there is one small issue with that, loading granularity, here it's two, I will have to make universal "load_long_misaligned" and pass load granularity to it. Still could be useful in future tho